### ISSUE TITLE
Not recognized as Switch Pro controller

### ISSUE THREAD

**Issue by Invictaz** [juan518munoz/PicoSwitch-WirelessGamepadAdapter#12]
There is no easy way to remap in the code without compiling and no WebUI to change anything.
So it would be best to output as a real Switch Pro controller

https://private-user-images.githubusercontent.com/20338545/311896486-39541a05-6089-43ed-ac52-9b89e27d54c0.jpg

**Comment by juan518munoz** [comment]
Does emulating as a pro controller allow the possibility to remap buttons with the switch UI? If so, please provide a video demonstrating the procedure.

**Comment by bmelanman** [comment]
Hello! I stumbled across this issue while looking around for rumble info (see issue #13 if interested) and I just so happened to be in a position to test this. 

**TL;DR: I can confirm that proper emulation of a Pro controller enables remapping in the Switch UI (and the remapping actually works, too!).**

I used a copy of [retro-pico-switch](https://github.com/DavidPagels/retro-pico-switch) that I modified to use [picow_ble_controller](https://github.com/intGus/picow_ble_controller), which allowed me to connect an Xbox controller to my switch. The retro-pico-switch project does whatever trickery is necessary for the Switch to think it's a legit Pro controller, which enables remap support. I remapped the screenshot button to be the home button, and there seemed to be no issues.

I spent ~30 mins to see if I could easily modify the `SwitchDescriptors.h` file hoping for a trivial fix, but unfortunately I have no clue what I'm doing, so someone else will have to figure the rest out (_or wait about a month and hope I remember to come back to this!_)

Let me know if anyone needs more info, hope this helps! :)

**Comment by Invictaz** [comment]
@bmelanman did you continue this?

**Comment by MegaCadeDev** [comment]
Thank you for the inspiration! This has been added to my fork!

**Comment by juan518munoz** [comment]
Interesting project @MegaCadeDev! Glad this helped you out 😄

**Comment by Invictaz** [comment]
@MegaCadeDev did you use the work of @bmelanman or https://github.com/intGus/picow_ble_controller for the proper Switch Pro emulation?

**Comment by MegaCadeDev** [comment]
> [@MegaCadeDev](https://github.com/MegaCadeDev) did you use the work of [@bmelanman](https://github.com/bmelanman) or https://github.com/intGus/picow_ble_controller for the proper Switch Pro emulation?

I used Retro Pico Switch as the guide to add the handshake that was required to register as a Switch Pro controller and gave proper attribution in my updates!
